### PR TITLE
Build new docs when publishing tagged releases.

### PR DESCRIPTION
**Pull request by pgavlin**
These changes wire up a new script that clones
https://github.com/pulumi/docs, regenerates the appropriate parts of the
docs site, and opens a pull request with the changes. This script
executes on any build in which we publish packages that _also_ has a tag
present.

### CODE REVIEW

**Comment by pgavlin** [comment]
I've verified that this script works locally on OSX. @ellismg how would you propose we test this in CI? Just remove the `! -z "${TRAVIS_TAG:-}"` condition for a moment?

**Comment by pgavlin** [comment]
It is probably not too much work to generalize this for other repos.

File: scripts/build-docs.sh [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+readonly ROOT=$(dirname "${0}")/..
+
+if [[ "${TRAVIS_PUBLISH_PACKAGES:-}" == "true" ]] && [[ ! -z "${TRAVIS_TAG:-}" ]]; then
+    NPM_VERSION=$("${ROOT}/scripts/get-version")
+    VERSION=${NPM_VERSION#"v"}
+
+    echo "Building SDK docs for version ${VERSION}:"
+
+    # Clone the repo and fetch its dependencies.
+    # git clone "https://github.com/pulumi/docs.git" "$(go env GOPATH)/src/github.com/pulumi/docs"
```
Comment (ellismg) [inline scripts/build-docs.sh]: Did you mean to have this commented out?  I'm guessing not and it was an artifact of you testing things locally.

**Comment by ellismg** [comment]
> I've verified that this script works locally on OSX. @ellismg how would you propose we test this in CI? Just remove the ! -z "${TRAVIS_TAG:-}" condition for a moment?

Yeah, seems reasonable.

File: scripts/build-docs.sh (hunk 1 above)
Comment (pgavlin) [reply scripts/build-docs.sh]: Fixed.

**Comment by pgavlin** [comment]
Here's the successful test CI job: https://travis-ci.com/pulumi/pulumi/builds/125999126
And here's the resulting (closed) PR: https://github.com/pulumi/docs/pull/1691

**Comment by pgavlin** [comment]
This should be good-to-go now that https://github.com/pulumi/docs/pull/1692 has landed.

**Comment by pgavlin** [comment]
Note that I think we will shortly supersede this with https://github.com/pulumi/scripts/pull/82

**Pull request merged**